src/api/define.js

```diff
@@ -122,13 +133,16 @@ export default function (name, opts) {
   Ctor[$props] = createInitProps(Ctor);
   Ctor[$renderer] = createRenderer(Ctor);
 
-  if (customElementsV1) {
-    window.customElements.define(uniqueName, Ctor, { extends: Ctor.extends });
-  } else if (customElementsV0) {
-    return document.registerElement(uniqueName, Ctor);
-  } else {
-    throw new Error('Skate requires native custom element support or a polyfill.');
+  let res;
+  try {
+    res = registerElement(uniqueName, Ctor);
+  } catch(e) {
+    uniqueName = generateUniqueName(name);
```

Comment:
this assumes that the 2nd call of `generateUniqueName(name)` will produce something unique. Probably safe, but would be safer to use something like @treshugart 's [idea/function](https://github.com/skatejs/skatejs/issues/723#issuecomment-239343459) until a unique name is found.